amqp/src/main/scala/akka/stream/alpakka/amqp/impl/AmqpFlow.scala

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.amqp.impl
+
+import akka.Done
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.stream._
+import akka.stream.alpakka.amqp.{AmqpWriteSettings, WriteMessage, WriteResult}
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler, OutHandler, StageLogging}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{Future, Promise}
+import scala.util.{Failure, Success, Try}
+
+/**
+ * Internal API.
+ *
+ * Base stage for AMQP flows that don't use asynchronous confirmations.
+ */
+@InternalApi private sealed abstract class AbstractAmqpFlowStage[T](
+    override val settings: AmqpWriteSettings,
+    promise: Promise[Done],
+    shape: FlowShape[WriteMessage[T], WriteResult[T]]
+) extends GraphStageLogic(shape)
+    with AmqpConnectorLogic
+    with StageLogging {
+
+  private def in = shape.in
+  private def out = shape.out
+
+  override def whenConnected(): Unit = ()
+
+  setHandler(
+    in,
+    new InHandler {
+      override def onUpstreamFailure(ex: Throwable): Unit = {
+        promise.failure(ex)
+        super.onUpstreamFailure(ex)
+      }
+
+      override def onUpstreamFinish(): Unit = {
+        promise.success(Done)
+        super.onUpstreamFinish()
+      }
+
+      override def onPush(): Unit =
+        publish(grab(in))
+    }
+  )
+
+  protected def publish(message: WriteMessage[T]): Unit
+
+  setHandler(out, new OutHandler {
+    override def onPull(): Unit =
+      if (!hasBeenPulled(in)) tryPull(in)
+  })
+
+  override def postStop(): Unit = {
+    promise.tryFailure(new RuntimeException("Stage stopped unexpectedly."))
+    super.postStop()
+  }
+
+  override def onFailure(ex: Throwable): Unit = {
+    promise.tryFailure(ex)
+    super.onFailure(ex)
+  }
+}
+
+/**
+ * Internal API.
+ *
+ * Simplest AMQP flow. Published messages without any delivery guarantees. Result is emitted as soon as message is
+ * published. Because there it's not verified whether message is confirmed or rejected, results are emitted always
+ * as confirmed. Since result is always confirmed, it would also make sense to emit just plain `passThrough` object
+ * instead of complete [[WriteResult]] (possibly it would be less confusing for users), but [[WriteResult]] is used
+ * for consistency with other variants and to make the flow ready for any possible future [[WriteResult]] extensions.
+ */
+@InternalApi private[amqp] final class AmqpSimpleFlow[T](writeSettings: AmqpWriteSettings)
```

Comment:
Please create different files for the implementations.

Reply:
Done.